x/shared/session.go

```diff
@@ -106,3 +107,45 @@ func GetProofWindowCloseHeight(sharedParams *sharedtypes.Params, queryHeight int
 	return GetProofWindowOpenHeight(sharedParams, queryHeight) +
 		int64(sharedParams.GetProofWindowCloseOffsetBlocks())
 }
+
+// GetEarliestClaimCommitHeight returns the earliest block height at which a claim
+// for the session that includes queryHeight can be committed for a given supplier
+// and the passed sharedParams.
+func GetEarliestClaimCommitHeight(
+	sharedParams *sharedtypes.Params,
+	queryHeight int64,
+	claimWindowOpenBlockHash []byte,
+	supplierAddr string,
+) int64 {
+	claimWindowOpenHeight := GetClaimWindowOpenHeight(sharedParams, queryHeight)
+
+	// Generate a deterministic random (non-negative) int64, seeded by the claim
+	// window open block hash and the supplier address.
+	randomNumber := poktrand.SeededInt63(claimWindowOpenBlockHash, []byte(supplierAddr))
```

Comment:
A claim is already ted to a single supplier. WHy do we need to add the other variable for entropy as well? I feel like we can reduce a lot of variables being passed around if so.

Reply:
I'm not 100% sure I followed the question.

The rationale here is that `claimWindowOpenBlockHash` will be common to all sessions (app/supplier pairings) with the same session number; therefore, in order for this offset to have the intended effect of distributing message commit heights across their respective windows, it must also consider additional entropy which is either unique to suppliers or applications. If we choose the former (supplier), relayminers can continue to batch claim and proof messages on a per session number basis, which would no longer be possible in the other case. If you're suggesting to use the claim itself as a source of entropy for this, I think that would create the same issue. I think the fact that relayminers can batch claims/proofs is basically all upside.

Reply:
Got it. 

I interpreted it as as `claimHash` which (IMO) should work even though you said `I think that would create the same issu`.

Now that I understand the reasoning, it makes sense. Contemplating if another name would make it clearer, but I think it's on me for misreading.